src/main/java/org/elasticsearch/transport/local/LocalTransportChannel.java

```diff
@@ -36,14 +36,16 @@
 public class LocalTransportChannel implements TransportChannel {
 
     private final LocalTransport sourceTransport;
+    private final TransportServiceAdapter sourceTransportServiceAdapter;
     // the transport we will *send to*
     private final LocalTransport targetTransport;
     private final String action;
     private final long requestId;
     private final Version version;
 
-    public LocalTransportChannel(LocalTransport sourceTransport, LocalTransport targetTransport, String action, long requestId, Version version) {
+    public LocalTransportChannel(LocalTransport sourceTransport, TransportServiceAdapter sourceTransportServiceAdapter, LocalTransport targetTransport, String action, long requestId, Version version) {
```

Comment:
All of this constructor handovers would not be necessary if the getter in the `Transport` interface as well... anything speaking against this (except keeping this one small and getting it in)